### PR TITLE
[DOCS] Create documentation for fields.yml (#6049)

### PR DESCRIPTION
**Pull request by bestpath-gb**
Add a new documentation page for `_meta/fields.yml`. This mentions what I'd consider most of the commonly used options, with a few examples.

### CODE REVIEW

**Comment by dedemorton** [comment]
Thanks for your contribution.

WRT writing style, we prefer to avoid passive voice when possible (active verbs are easier to read). Also, you should add line breaks before your code sections (for example, before line 7). We're migrating to asciidoctor, which is less permissive than asciidoc, so it's best to be on the safe side.

I'd like to offer more editorial feedback, but I can't right now. If this doesn't get merged before I come back to it, I'll submit a follow-up edit. Thanks again for doing this!

File: docs/devguide/fields-yml.asciidoc [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  title: mybeat
+  description: These are the fields used by mybeat.
+  fields:
+    - name: surname
+      type: keyword
+      required: true
+      copy_to: full_name
+      description: >
+        The surname.
+      multi_fields: <1>
```
Comment (faec) [inline docs/devguide/fields-yml.asciidoc]: Both `surname` and `forenames` here are themselves `keyword` types, which would make this redundant, right? The description above also describes using this on text fields, not keywords. Perhaps this section should use a different example field that would more naturally be of type text.

**Comment by dedemorton** [comment]
jenkins test this please

File: docs/devguide/fields-yml.asciidoc (hunk 1 above)
Comment (dedemorton) [reply docs/devguide/fields-yml.asciidoc]: @bestpath-gb Will you be able to update the example as suggested here? I appreciate the effort that you put into creating this PR and would like to help you get this merged.

Comment (bestpath-gb) [reply docs/devguide/fields-yml.asciidoc]: Thanks for the feedback. Sorry for the delay in getting back to this. I'm hoping to review and update the examples this week.

**Comment by dedemorton** [comment]
I'm adding the `needs_edit` label to this PR so I can come back and edit the content after it's approved and merged.

File: docs/devguide/fields-yml.asciidoc (hunk 1 above)
Comment (bestpath-gb) [reply docs/devguide/fields-yml.asciidoc]: That was a great catch, @faec . I've updated the examples based on the Elasticsearch reference for those mapping parameters.

**Comment by dedemorton** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by dedemorton** [comment]
Sorry I haven't been able to edit this yet. I think it's best to merge the PR since it's technically correct. I'll follow up with edits in a separate PR. Just launched the doc build. When it passes, I will merge. Thanks again for your contribution. My apologies that it's taken me awhile to get back to this.

**Comment by dedemorton** [comment]
I've created a meta issue to track the content that requires an edit, so I'm removing the needs_edit
label.

**Pull request merged**